### ISSUE TITLE
What is the purpose of noWrite in rules/reactivity.ts?

### ISSUE THREAD

**Issue by aercolino** [solidjs-community/eslint-plugin-solid#35]
**What is the purpose of noWrite in rules/reactivity.ts?**

I see that it's [taken into account](https://github.com/solidjs-community/eslint-plugin-solid/blob/8372d356d4e27df0f14d20eed35ac81624e63054/src/rules/reactivity.ts#L417) but not [tested](https://github.com/solidjs-community/eslint-plugin-solid/blob/8372d356d4e27df0f14d20eed35ac81624e63054/test/rules/reactivity.test.ts).

I use a pattern to pass signals to children like this:

```js
function Parent(props) {
  let value, setValue;

  createRenderEffect(() => {
    [value, setValue] = createSignal(props.defaultValue); // <-- here
  });

  return <Child value={value}, setValue={setValue} />
}
```

And the noWrite rule gets triggered onto "value" there, but that's clearly not a problem because the reassignment is just the way to set value to the accessor and be able to use it outside of the createRenderEffect.

Should I setup things differently? (I'm new to Solidjs)

**Comment by aercolino** [comment]
I just found out a different setup, that works equally well:

```js
function Parent(props) {
  const [value, setValue] = createSignal(null);

  createRenderEffect(() => {
    setValue(props.defaultValue);
  });

  return <Child value={value}, setValue={setValue} />
}
``` 

However, the question stands.

**Comment by joshwilsonvu** [comment]
I appreciate you taking the time to look through the code! The big reason the rule asks you to make signals, etc. `const` is so that it can more reliably analyze what's going on. (It's also more idiomatic Solid code, emphasizing that `value` is a function.)

If you have a variable that is sometimes a signal and sometimes not, you're inevitably going to run into false positives/negatives with warnings because that's impossible track. We think that avoiding that is asking you to make certain variables `const`. 

As an aside, your first example has problems that your second example fixes. If `props.defaultValue` changed, the effect would rerun and reassign `value` to _a new signal_, rather than updating the signal's value. You'd get problems with stale references and that's almost certainly not what you want. In both examples, you'll likely want to call `value` in the JSX, so that the prop is the value, not the signal function itself.

Ideally, you might want to write that pattern as the following. 

```jsx
function Parent(props) {
  const [value, setValue] = createSignal(props.defaultValue);

  return <Child value={value()} setValue={setValue} />
}
```

Thanks for the question!

**Comment by goenning** [comment]
hey @joshwilsonvu I have a similar code as suggested above, but I'm getting a rule warning on it. Is this expected?

<img width="831" alt="Screen Shot 2022-10-31 at 05 18 56" src="https://user-images.githubusercontent.com/94755/198962901-0dc5aa3b-47cc-4026-9a06-4fe081324904.png">

**Comment by joshwilsonvu** [comment]
Yes, but you can & should ignore those warnings with `// eslint-disable-next-line solid/reactivity` as long as you understand that if those values change, `createSignal` won't react to it. 

The rule is there to let you know about cases where you might expect Solid to react to a change in a prop/signal but it won't. Using props to initialize a signal is a fairly common case where this is okay, so there's an escape hatch where if your prop name starts with `initial`/`default` it won't warn, because you've shown that you intend to ignore any changes. For other cases like yours ignoring the warning accomplishes the same thing.
